File: hardening/month_offset_oracle.php
<?php
// Independent-runtime oracle: applies one relative modifier with PHP's DateTime::modify and prints one ISO date.
// Usage: php month_offset_oracle.php '<yyyy-mm-dd>' '<relative modifier>'
$date = new DateTime($argv[1]);
$date->modify($argv[2]);
echo $date->format('Y-m-d'), "\n";

File: hardening/month_offset_oracle.rb
# Independent-runtime oracle: shifts a date by whole months with Ruby's Date#>> and prints one ISO date.
# Usage: ruby month_offset_oracle.rb '<yyyy-mm-dd>' '<months>'
require 'date'
puts (Date.iso8601(ARGV[0]) >> ARGV[1].to_i).to_s
